### PR TITLE
More refactoring: use a common interface for all Lokid requests

### PR DESCRIPTION
**Pull request by msgmaxim**
Again, mostly moving code around. This brings us one step closer to enabling username/password to talk to Lokid.

### CODE REVIEW

File: httpserver/service_node.cpp [hunk 1]
```diff
@@ -197,12 +197,12 @@ ServiceNode::ServiceNode(boost::asio::io_context& ioc,
                          boost::asio::io_context& worker_ioc, uint16_t port,
                          const loki::lokid_key_pair_t& lokid_key_pair,
                          const std::string& db_location,
-                         uint16_t lokid_rpc_port)
+                         LokidClient& lokid_client, uint16_t lokid_rpc_port)
```
Comment (BeaudanBrown) [inline httpserver/service_node.cpp]: `lokid_rpc_port` is no longer being used in service_node.cpp

File: httpserver/service_node.cpp [hunk 2]
```diff
@@ -519,14 +571,26 @@ make_batch_requests(std::vector<std::string>&& data) {
...
-        ioc_, lokid_rpc_port_, lokid_key_pair_, params,
-        [cb = std::move(cb)](const std::string& body_str) {
-            const json body = json::parse(body_str, nullptr, false);
+
+    nlohmann::json params;
+
+    params["max_height"] = test_params.max_height;
+    params["seed"] = test_params.seed;
+
+    auto on_resp =
```
Comment (BeaudanBrown) [inline httpserver/service_node.cpp]: My clang format lowers the indentation

File: httpserver/service_node.cpp (hunk 1 above)
Comment (msgmaxim) [reply httpserver/service_node.cpp]: removed

File: httpserver/service_node.cpp (hunk 2 above)
Comment (msgmaxim) [reply httpserver/service_node.cpp]: ran clang format

**Pull request merged**